connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java

```diff
@@ -70,55 +100,45 @@ public CorrelationResult correlate(
           "Failed to evaluate activation condition against the provided input", e);
     }
 
-    if (correlationPoint instanceof StartEventCorrelationPoint startCorPoint) {
-      return triggerStartEvent(definition, startCorPoint, variables);
-    } else if (correlationPoint instanceof MessageCorrelationPoint msgCorPoint) {
-      return triggerMessage(
-          definition,
-          msgCorPoint.messageName(),
-          msgCorPoint.correlationKeyExpression(),
+    return switch (correlationPoint) {
+      case StartEventCorrelationPoint ignored -> triggerStartEvent(
+          matchedElements,
+          activatedElement,
+          variables);
+      case MessageCorrelationPoint msgCorPoint -> triggerMessage(
+          matchedElements,
+          activatedElement,
           variables,
           resolveMessageId(msgCorPoint.messageIdExpression(), messageId, variables));
-    } else if (correlationPoint instanceof MessageStartEventCorrelationPoint msgStartCorPoint) {
-      return triggerMessageStartEvent(definition, msgStartCorPoint, variables);
-    } else if (correlationPoint
-        instanceof BoundaryEventCorrelationPoint boundaryEventCorrelationPoint) {
-      return triggerMessage(
-          definition,
-          boundaryEventCorrelationPoint.messageName(),
-          boundaryEventCorrelationPoint.correlationKeyExpression(),
-          variables,
-          resolveMessageId(
-              boundaryEventCorrelationPoint.messageIdExpression(), messageId, variables));
-    } else {
-      // this should never happen, thus not wrapped in a CorrelationResult
-      throw new ConnectorException(
-          "Process correlation point "
-              + correlationPoint.getClass()
-              + " is not supported by Runtime");
-    }
+      case MessageStartEventCorrelationPoint ignored -> triggerMessageStartEvent(
+          matchedElements,
+          activatedElement,
+          variables);
+    };
   }
 
   protected CorrelationResult triggerStartEvent(
-      InboundConnectorDefinitionImpl definition,
-      StartEventCorrelationPoint correlationPoint,
+      List<InboundConnectorElement> matchedElements,
```

Comment:
The idea was to handle the situation where multiple elements matched the activation condition and we only correlated to one of them. This way we can let the connector developer decide how to handle such a case. But TBH now I'd rather remove it and refuse to correlate in this situation and return a `Failure` response, since this is essentially a misconfiguration.

Reply:
I see, thanks for the explanation 👌